File: basic/satcomPy/GroundStation.py
import uuid
import threading

class GroundStation:
    def __init__(self, satellite, language='en'):
        self.satellite = satellite
        self.uuid = str(uuid.uuid4())
        self.language = language
        self.lock = threading.Lock()
        if language == 'cn':
            print(f"创建 GroundStation 对象，UUID 为: {self.uuid}")
        else:
            print(f"GroundStation created with UUID: {self.uuid}")

    def send_data(self):
        if self.language == 'cn':
            print(f"地面站 {self.uuid} 正在向卫星发送数据...")
        else:
            print(f"GroundStation {self.uuid} sending data to satellite...")
        self.satellite.notify_ground_stations()

    def receive_data(self):
        if self.language == 'cn':
            print(f"地面站 {self.uuid} 接收到来自卫星的数据。")
        else:
            print(f"GroundStation {self.uuid} received data from satellite.")

File: basic/satcomPy/GroundStation.java
import java.util.UUID;

public class GroundStation {
    private Satellite satellite;
    private String uuid;
    private String language;

    public GroundStation(Satellite satellite, String language) {
        this.satellite = satellite;
        this.language = language;
        this.uuid = UUID.randomUUID().toString();
        if (language.equals("cn")) {
            System.out.println("创建 GroundStation 对象，UUID 为: " + this.uuid);
        } else {
            System.out.println("GroundStation created with UUID: " + this.uuid);
        }
    }

    public void sendData() {
        if (language.equals("cn")) {
            System.out.println("地面站 " + this.uuid + " 正在向卫星发送数据...");
        } else {
            System.out.println("GroundStation " + this.uuid + " sending data to satellite...");
        }
        satellite.notifyGroundStations();
    }

    public void receiveData() {
        if (language.equals("cn")) {
            System.out.println("地面站 " + this.uuid + " 接收到来自卫星的数据。");
        } else {
            System.out.println("GroundStation " + this.uuid + " received data from satellite.");
        }
    }

    public String getUUID() {
        return this.uuid;
    }
}
